packages/test/service-load-test/src/faultInjectionDriver.ts

```diff
@@ -0,0 +1,173 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IEvent, ITelemetryBaseLogger } from "@fluidframework/common-definitions";
+import { assert, EventForwarder } from "@fluidframework/common-utils";
+import {
+    IDocumentDeltaConnection,
+    IDocumentDeltaConnectionEvents,
+    IDocumentDeltaStorageService,
+    IDocumentService,
+    IDocumentServiceFactory,
+    IDocumentStorageService,
+    IResolvedUrl,
+} from "@fluidframework/driver-definitions";
+import {
+    IClient,
+    ISummaryTree,
+    IErrorTrackingService,
+    IDocumentMessage,
+    INack,
+    NackErrorType,
+} from "@fluidframework/protocol-definitions";
+
+export class FaultInjectionDocumentServiceFactory implements IDocumentServiceFactory {
+    private  readonly _last = new Map<IResolvedUrl, FaultInjectionDocumentService>();
+
+    public get protocolName() { return this.internal.protocolName;}
+    public get documentServices() {return this._last;}
+
+    constructor(private readonly internal: IDocumentServiceFactory) { }
+
+    async createDocumentService(resolvedUrl: IResolvedUrl, logger?: ITelemetryBaseLogger): Promise<IDocumentService> {
+        const internal = await this.internal.createDocumentService(resolvedUrl, logger);
+        const ds = new FaultInjectionDocumentService(internal);
+        assert(!this._last.has(resolvedUrl), "one ds per resolved url instance");
+        this._last.set(resolvedUrl, ds);
+        return ds;
+    }
+    async createContainer(
+        createNewSummary: ISummaryTree,
+        createNewResolvedUrl: IResolvedUrl,
+        logger?: ITelemetryBaseLogger):
+        Promise<IDocumentService> {
+        return this.internal.createContainer(
+            createNewSummary,
+            createNewResolvedUrl,
+            logger,
+        );
+    }
+}
+
+export class FaultInjectionDocumentService implements IDocumentService {
+    private  _lastDeltaStream: FaultInjectionDocumentDeltaConnection | undefined;
```

Comment:
I think "current" is clearer, if I'm reading the code right.

Reply:
yup. class when through a couple iterations and just needs cleanup